persistence/pg.go

```diff
@@ -186,9 +178,11 @@ func (p *PostgresPersister) MarkHtlcSettled(ctx context.Context,
 			HtlcID: key.HtlcID,
 		}
 
-		// Check to see if htlc exists.
-		err = tx.ModelContext(ctx, &htlc).
+		// Check to see if htlc exists. Use FOR UPDATE to prevent a concurrent
+		// call for the same htlc to execute in parallel.
+		err := tx.ModelContext(ctx, &htlc).
 			WherePK().
+			For("UPDATE").
```

Comment:
Postgres locking is not easy. Maybe the `Get` function also isn't completely safe if an update comes in between the two select statements in there. Might need to use repeatable read isolation.

Reply:
Maybe repeatable read doesn't require `FOR UPDATE` then here?

Reply:
Regarding the safety, I agree with you, maybe we can acquire the lock for the invoice before acquiring the lock for the htlc? 

So we should not have problems?

Reply:
Or

We could SELECT FOR UPDATE all htlcs linked to the invoice (But I really don't like this idea)